### PR TITLE
feat(android): Bundle GITHUB_SHA into Android client

### LINKED ISSUE #6400: Android: Bundle git sha into build and show in settings
This will answer the question of, "what release is published?" in an automated way.

### PR DESCRIPTION
**Pull request by jamilbk**
Closes #6400 

Screenshot 2024-08-21 at 11 24 16 PM

### CODE REVIEW

File: kotlin/android/app/build.gradle.kts [hunk 1]
```diff
@@ -59,6 +59,9 @@ android {
         versionName = "1.2.1"
         multiDexEnabled = true
         testInstrumentationRunner = "dev.firezone.android.core.HiltTestRunner"
+
+        val gitSha = System.getenv("GITHUB_SHA") ?: "unknown"
```
Comment (thomaseizinger) [inline kotlin/android/app/build.gradle.kts:63]: What about builds that don't happen on CI? Where are the builds made that get submitted to the play store for review?

**Review by thomaseizinger — COMMENTED** [review]
`git-describe` is made for this but it is output is much nicer if you have a tag it can use.

File: kotlin/android/app/build.gradle.kts (hunk 1 above)
Comment (jamilbk) [reply kotlin/android/app/build.gradle.kts:63]: It'll be `unknown`. Playstore builds always happen via the Kotlin workflow in CI.

**Comment by jamilbk** [comment]
> `git-describe` is made for this but it is output is much nicer if you have a tag it can use.

Yeah I think I would be open to tagging these if there was automation, otherwise we'll inevitable run into cases where the tag doesn't match the published release. It seems like it could be possible using the Publishing API and some automation: https://developers.google.com/android-publisher/api-ref/rest

**Comment by thomaseizinger** [comment]
> > `git-describe` is made for this but it is output is much nicer if you have a tag it can use.
> 
> Yeah I think I would be open to tagging these if there was automation, otherwise we'll inevitable run into cases where the tag doesn't match the published release. It seems like it could be possible using the Publishing API and some automation: https://developers.google.com/android-publisher/api-ref/rest

I found this: https://github.com/marketplace/actions/upload-android-release-to-play-store

It is also referenced in a couple of blog posts on automating Android releases.

How about this:

- Make draft releases like we do for the other artifacts and attach the APKs to them
- Trigger the above workflow when a release gets published (https://docs.github.com/es/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release)

**Comment by jamilbk** [comment]
> > > `git-describe` is made for this but it is output is much nicer if you have a tag it can use.
> 
> > 
> 
> > Yeah I think I would be open to tagging these if there was automation, otherwise we'll inevitable run into cases where the tag doesn't match the published release. It seems like it could be possible using the Publishing API and some automation: https://developers.google.com/android-publisher/api-ref/rest
> 
> 
> 
> I found this: https://github.com/marketplace/actions/upload-android-release-to-play-store
> 
> 
> 
> It is also referenced in a couple of blog posts on automating Android releases.
> 
> 
> 
> How about this:
> 
> 
> 
> - Make draft releases like we do for the other artifacts and attach the APKs to them
> 
> - Trigger the above workflow when a release gets published (https://docs.github.com/es/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release)

How does this solve the review rejection possibility though? It happens often enough that we should account for it.

We need it to happen the other way around. When the app is approved, publish its release and attach the APK.

**Comment by thomaseizinger** [comment]
> How does this solve the review rejection possibility though? It happens often enough that we should account for it.
> 
> We need it to happen the other way around. When the app is approved, publish its release and attach the APK.

It doesn't directly. Could we just treat such version numbers as burnt and make a tag after? We can still merge the changelogs and stuff into the new one.

**Review by thomaseizinger — APPROVED** [review]

**Comment by jamilbk** [comment]
> It doesn't directly. Could we just treat such version numbers as burnt and make a tag after? We can still merge the changelogs and stuff into the new one.

Hmm let's move discussion to #3321

**Pull request merged**